docs/code-organization/git-quickstart.mdx

```diff
@@ -0,0 +1,331 @@
+you can just re-create the commit with the `--amend` flag!
+
+This will replace the latest commit with the new one that hopefully is fine
+this time. ;)
+
+```
+$ git commit --amend -m "docs: Fix typo in Example docs" -m "Example should be spelled with a single e rather than two es.
+dquote>
+dquote>[SCSE-24]"
+[main f253a79] docs: Fix typo in Example docs
+ Date: Thu Oct 13 23:39:33 2022 +0800
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+```
+
+:::caution
+Note that by doing this, you are recreating history!
+NEVER ever re-create history that is already public and in the `main` branch as
+it will force everyone to change their history as well.
+:::
+
+## Branching
+
+You should definitely create a branch for what you are working on before you
+start working on your super cool feature.
+
+To quote [Pro Git Chapter 3][pro-git-chap-3], branching means you diverge from
+the main line of development and continue to do work without messing with
+that main line.
+
+The [`git switch`][git-switch] command has been added in a recent update of Git
+to help make the process simpler. `-c` will create a branch with the specified
+name while not providing it will simply switch you over to that branch.
+
+```
+$ git switch -c docs/SCSE-24
+Switched to a new branch 'docs/SCSE-24'
+
+$ git switch main
+Switched to branch main
+
+$ git switch -c docs/SCSE-24
+fatal: a branch named 'docs/SCSE-24' already exists
+```
+
+## Pulling/Pushing Your Changes
+
+You've finally finished that super cool feature you worked on! Now you just
+need to **push** it to GitHub where we host our code for everyone to see it!
+
+However, you should definitely check that your changes are compatible with all
+the new changes other people have finished while you were working on your
+project.
+
+To do so, apply your changes on top of the current code on GitHub with [`git
+pull --rebase`][git-pull].
+
+```
+$ git pull --rebase origin main
+remote: Enumerating objects: 204, done.
+remote: Counting objects: 100% (204/204), done.
+remote: Compressing objects: 100% (112/112), done.
+remote: Total 204 (delta 104), reused 181 (delta 91), pack-reused 0
+Receiving objects: 100% (204/204), 72.83 KiB | 298.00 KiB/s, done.
+Resolving deltas: 100% (104/104), completed with 25 local objects.
+From github.com/ntuscse/docs
+[More output omitted]
+```
+
+After checking your code still works or after you finish fixing it, it's time
+to show it to everyone by pushing it to GitHub with [`git push`][git-push]!
+
+```
+$ git push
+Total 204 (delta 104), reused 181 (delta 91), pack-reused 0
+remote:
+remote: Create a pull request for 'docs/SCSE-24' on GitHub by visiting:
+remote:      https://github.com/ntuscse/docs/pull/new/docs/SCSE-24
+remote:
+To github.com/ntuscse/docs
+ * [new branch]      docs/SCSE-24 -> docs/SCSE-24
+```
+
+If you forgot to create a branch, you can still specify what the name at GitHub
+should be. In the following example, we assume you worked on `main` and the
```

Comment:
Let's be more verbose, by saying i.e. "for example, if you **accidentally** worked on the main branch locally, and wish to move your commits to your task branch xyz...", or something to that effect